lua/cmp_luasnip/choice.lua

```diff
@@ -0,0 +1,45 @@
+local cmp = require("cmp")
+
+local source = {}
+
+source.new = function()
+	return setmetatable({}, { __index = source })
+end
+
+source.get_keyword_pattern = function()
+	return "\\%([^[:alnum:][:blank:]]\\|\\w\\+\\)"
+end
+
+function source:is_available()
+	return require('luasnip.session').active_choice_node ~= nil
+end
+
+
+function source:get_debug_name()
+	return "luasnip_choice"
+end
+
+function source:complete(_, callback)
+  local items = {}
+
+  for idx,word in pairs(require('luasnip').get_current_choices()) do
+    table.insert(items, {
+      word = word,
```

Comment:
It would be sick if we could just switch to the choice temporarily instead of inserting some text, but I don't think it's possible to change the `"hover"`-action from simply inserting the `word` to something more complicated.

As it stands, I think we should not insert any text (eg `word = ""`).